substrate/frame/staking-async/precompiles/interface.sol

```diff
@@ -0,0 +1,188 @@
+* Allow EVM contracts to experiment building custom pooled-staking (ala Lido).
+* Allow existing parachains to (possibly) re-implement their protocol in AH (Bifrost vDOT, Acala LDOT, moonbeam stDOT).
+* (experimental) A validator can put their funds into a contract that interfaces with the following. This contract is fully owned by the validator and they can do anything, except it restricts them from certain arbitrary commission changes. Require the EVM account to be able to transact with other pallets (e.g. validator needs to set session keys, claim rewards etc), and a good template contract for it.
+
+## Implementation Notes
+
+### Upgradability and Splitting
+Precompiles are not upgradable in any meaningful way, yet we can later add more features to each, as an extension to that particular interface. This is also a motivation for the interfaces to be broken apart as much as possible, rather than one massive one, beyond a better exercise of separation of concerns. For example, I opted to not include the "blocking" mechanism that validator have access to in the interface (tentative, open to suggestions), and if need be it can be added later in a new precompile (`StakingRolesV2`) that extends the current oen (`StakingRoles`).
+
+### Bond Limits
+
+TODO
+
+## Risks / Stability
+
+TODO
+
+## Resources
+
+* https://research.lido.fi/t/updated-lido-kusama-polkadot-ls-by-mixbytes/877 / https://research.lido.fi/t/sunsetting-of-lido-on-polkadot-and-kusama/4067
+* https://github.com/moonbeam-foundation/moonbeam/blob/0e600693f70b67c59ed6a9688deb91fa5339cd5a/precompiles/parachain-staking/StakingInterface.sol
+
+*/
+
+/// @title Staking Interface
+/// @notice Allows users to bond, unbond, and manage their bonded amounts in the staking system. For doing more interesting things, such as nomination, see `StakingRoles`.
+/// @dev Talks to the directly `pallet-staking-async`.
+/// @author kianenigma
+interface Staking {
+	struct Stake {
+		uint128 active; // The amount of tokens actively bonded.
+		uint128 total; // The total tokens bonded, active + unbonding.
+	}
+
+	struct Unbonding {
+		uint32 era; // The era when the unbonding will be available.
+		uint128 amount; // The amount of tokens that are unbonding.
+	}
+
+	/// @notice Bonds the caller's account, increasing the bonded amount by up to `value` tokens. The caller's bonded amount after this call must be be greater than `minimum_bond()`.
+	/// @param value The amount of tokens to bond.
+	/// @return amount The actual amount that was additionally bonded. Returns 0 on failure.
+	/// @dev At the moment, staking pallet only imposes `ED` as the minimum bond. A separate PR should parameterize this and use `ed.max(minimum_bond)` instead. This is a wrapper for `bond` and `bond_extra`.
+	function bond(uint128 value) external returns (uint128 amount);
```

Comment:
I think this function should be `payable`, instead of taking `uint128 value` as an argument (I assume the user (or contract) bonds with their native token balance like DOT, KSM, WND, etc).